src/wip/streaming-dca.rain

```diff
@@ -0,0 +1,188 @@
+        value: 0.5.1
+        description: Amount of RED tokens sold per hour.
+      - label: RED sold per day
+        value: 0.5.2
+        description: Amount of RED tokens sold per day.
+      - label: Minimum bounty
+        value: 0.5.3
+        description: Minimum bounty offered in BLUE tokens.
+      - label: Maximum bounty
+        value: 0.5.4
+        description: Maximum bounty offered in BLUE tokens.
+      - label: Bounty epoch
+        value: 0.5.5
+        description: Bounty epoch time in seconds
+      - label: Bounty epoch increase
+        value: 0.5.6
+        description: Increase in bounty per epoch
+      - label: Minimum ratio
+        value: 0.5.7
+        description: Maximum bounty offered in BLUE tokens.
+      
+                     
+---
+#input-token-address !Order input token address.
+#output-token-address !Order output token address.
+
+#twap-io-fee !Twap fee for the distribution token pool, paired with either stable token. 
+#min-ratio !Minimum input ouput ratio for the order.
+
+#bounty-min !Minimum bounty to offer for each trade in USD.
+#bounty-epoch-increase !Amount to increase bounty per bounty epoch.
+#bounty-epoch !Bounty epoch in seconds.
+#bounty-max !Maximum bounty that can be offered. 
+
+#per-second-rate !The rate of stable denominated distributed tokens distributed per second.
+#strat-strat-time !Strategy start timestamp
+#get-last-time !Binding to get last time and current time
+#plottables !Binding for charting metrics.
+
+#uniswap-words !The subparser for the Uniswap words
+#orderbook-subparser !The subparser for the Orderbook words
+#last-time-key "last-time-key"
+
+#bounty-auction
+  time-since-cooldown: ,
+  bounty: min(
+    linear-growth(
+      bounty-min
+      bounty-epoch-increase
+      div(time-since-cooldown bounty-epoch) 
+    )
+    bounty-max
+  );
+
+#get-last-time-plot
+ current-time: block-timestamp(),
+ last-time: strat-strat-time;
+
+#get-last-time-prod
+ current-time: block-timestamp(),
+ last-time: any(
+    get(hash(order-hash() last-time-key))
+    strat-strat-time
+  );
+
+#calculate-budget-input
+  current-budget: ,
+  _: uniswap-v3-quote-exact-input(
+    output-token-address input-token-address
+    current-budget twap-io-fee
+  );
+
+#plottables-plot
+  _: per-second-rate,
+  tokens-per-hour: mul(per-second-rate 3600),
+  tokens-per-day: mul(per-second-rate 86400),
+  _: bounty-min,
+  _: bounty-max,
+  _: bounty-epoch,
+  _: bounty-epoch-increase,
+  _: min-ratio;
+
+#plottables-prod
+ :;
+  
+#calculate-io
+  using-words-from uniswap-words orderbook-subparser
+  
+  current-time last-time: call<'get-last-time>(), 
+  elapsed-time: saturating-sub(current-time last-time),
+  current-budget: mul(per-second-rate elapsed-time),
+  bounty-amount: call<'bounty-auction>(elapsed-time),
+  input-amount: call<'calculate-budget-input>(current-budget),
+  max-amount: current-budget,
+  io-ratio: div(saturating-sub(input-amount bounty-amount) current-budget),
```

Comment:
you've got a sat sub here on the numerator, is an io ratio of `0` actually allowed?